### PR TITLE
Added Fracktal Works printers

### PR DESCRIPTION
**Pull request by ganeshmev**
Added Fracktal Works printers

### CODE REVIEW

**Comment by konskarm** [comment]
Hi @ganeshmev. I have a few things to mention from a quick scan:

- This is a **huge** amount of files. They can be reduced significantly by taking advantage of our inheritance system. You can make base printer profiles and then inherit them in your printers to avoid duplicating settings among them. Also, then you will be able to attach qualities to base profiles, so every printer of yours will inherit them by default. Have a look how creality printers are structured.
- There are extruders for a "volterra_dual" machine, which does not exist in the PR.
- The settings_version needs to be updated to 16.
- There are mixed tabs and spaces in the .def.json files. This leads to the files not being read properly. They have to be changed so that they contain **only** spaces.
- The "category" field is deprecated and should not appear in the metadata of definition files.
- The extruder "extruders/julia_extended" does not have any extension (should be .def.json)
- There are settings in qualities which do not exist in Cura (in fdmprinter.def.json that is).
- ..and more

Please have a look at the failing tests and fix all the issues reported there.

**Comment by nallath** [comment]
Could you also actually use descriptive commit messages? We try to follow the principles explained in https://chris.beams.io/posts/git-commit/ as much as possible.

**Comment by ganeshmev** [comment]
Can anyone help me with the errors?
I don't understand them.
Please help me in solving them.


From my first build to latest, I have done the below.
1. setting_version = 16
I was using previous version as I was running older version of Cura! 
Migrating to newer build, I had not changed to newer version. I made this change on all the files.

2. Category
I had machine Category defined. Now as its removed and following it as per Cura.

3. Extruders
Had previously kept all the machines-extruders in separate per folders. Now its kept in root of extruders. 

4. Few changes in definition. 
Previously my cura was built on @smartavionics build. 
So I was using the below
"bridge_over_support":{"value": true},
"bridge_skin_min_area":{"value": 0},
"wall_overhang_over_support" : {"value": true},

Now, as Cura doesn't support them, I had to delete/ stop the usage of them.

5. "support_infill_angle"-renamed ("support_infill_angle" changed to "support_infill_angles")
As the definition name was changed, I had to follow them. 


Thank you.

**Comment by nallath** [comment]
You seem to have invalid characters in your JSON (so the files are invalid).

Please test the files locally before committing them!

**Comment by ganeshmev** [comment]
> You seem to have invalid characters in your JSON (so the files are invalid).
> 
> Please test the files locally before committing them!

Its working locally. Cura is working
I suppose its something else.

**Comment by fieldOfView** [comment]
You made an invalid change to fdmprinter.def.json:
https://github.com/Ultimaker/Cura/pull/8650/files#diff-835d0333a273b72a90298b1d2ccecee98f4e9633e2518d2f56e28ee39b515b99R4403

**Comment by nallath** [comment]
> > You seem to have invalid characters in your JSON (so the files are invalid).
> > Please test the files locally before committing them!
> 
> Its working locally. Cura is working
> I suppose its something else.

It's invalid JSON. Unless you have re-implemented the json parser locally to accept this, it will also fail locally. I've pulled in your changes and when I start Cura, I get error messages that the file is invalid. 

So no, you didn't test this. It's fine if you forgot it, but please don't lie about it.

**Comment by ganeshmev** [comment]
> You made an invalid change to fdmprinter.def.json:
> https://github.com/Ultimaker/Cura/pull/8650/files#diff-835d0333a273b72a90298b1d2ccecee98f4e9633e2518d2f56e28ee39b515b99R4403
> 
> And the same error is in fusedform_base.def.json

Oh Thanks.
I shall recheck all the JSON again.

**Comment by ganeshmev** [comment]
> > You made an invalid change to fdmprinter.def.json:
> > https://github.com/Ultimaker/Cura/pull/8650/files#diff-835d0333a273b72a90298b1d2ccecee98f4e9633e2518d2f56e28ee39b515b99R4403
> > And the same error is in fusedform_base.def.json
> 
> Oh Thanks.
> I shall recheck all the JSON again.



> > > You seem to have invalid characters in your JSON (so the files are invalid).
> > > Please test the files locally before committing them!
> > 
> > 
> > Its working locally. Cura is working
> > I suppose its something else.
> 
> It's invalid JSON. Unless you have re-implemented the json parser locally to accept this, it will also fail locally. I've pulled in your changes and when I start Cura, I get error messages that the file is invalid.
> 
> So no, you didn't test this. It's fine if you forgot it, but please don't lie about it.

@nallath, I really appreciate your work, Thank you.
I have been stuck with [this issue](https://github.com/Ultimaker/Cura/issues/7580) and yet finding a solution.

So, I made changes on cura which was installed and took JSON/made upon it. So, in that I have missed few which I had modded.

**Comment by fieldOfView** [comment]
Please don't remove required files. Cura will not work without fdmextruder.def.json

**Comment by nallath** [comment]
Also, why are you making changes to fdmprinter? This will result in changes for _all_ other printers.

File: resources/definitions/julia_extended.def.json [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        "preferred_quality_type": "normal",
+        "has_variants": true,
+        "variants_name": "Nozzle",
+        "preferred_variant_name": "Model 0.4 mm",
+        "preferred_material": "fracktal_pla_175",
+        "machine_extruder_trains": {
+            "0": "julia_extended_extruder_0"
+        },
+        "visible": true,
+        "machine_variant_shortcode": "JE"
```
Comment (Ghostkeeper) [inline resources/definitions/julia_extended.def.json:22]: What's this? This metadata is not used by Cura in any case.

**Comment by Ghostkeeper** [comment]
A lot of these profiles are exactly the same, and just copied from each other. For instance, see the file `resources/quality/julia_basic/ABS/Model Nozzles/julia_basic_model_0.25_abs_high.inst.cfg`:

```
[general]
version = 4
name = High
definition = julia_basic

[metadata]
setting_version = 16
type = quality
quality_type = high
weight = 1
material = fracktal_abs_175
variant = Model 0.25 mm

[values]
layer_height = 0.1
brim_width = 8
cool_fan_enabled = False
cool_min_speed = 6
support_fan_enable = False
bridge_fan_speed = 0
wall_overhang_fan_speed = 0
```

And then `resources/quality/julia_extended/ABS/Model Nozzles/julia_extended_model_0.25_abs_high.inst.cfg`:

```
[general]
version = 4
name = High
definition = julia_extended

[metadata]
setting_version = 16
type = quality
quality_type = high
weight = 1
material = fracktal_abs_175
variant = Model 0.25 mm

[values]
layer_height = 0.1
brim_width = 8
cool_fan_enabled = False
cool_min_speed = 6
support_fan_enable = False
bridge_fan_speed = 0
wall_overhang_fan_speed = 0
```

This can be reduced by simply setting the `quality_definition` metadata of `julia_extended` to `julia_basic`. See the Ultimaker 3 vs. Ultimaker 3 Extended printers for an example. You can still override settings if there is a difference. If that is done, I expect that you'll be able to reduce the number of extruders 7 or 8-fold (since there are 8 printers here; I didn't check if all of them have the same profiles though). As it stands, this change adding 1300 profiles to our current 2300 is not going to be accepted due to the impact on the start-up time for all users.

**Comment by nallath** [comment]
Friendly reminder that there are a number of review comments still left open. If they are not handled, we will (eventually) close this PR.

**Comment by nallath** [comment]
Closing this due to inactivity

**Comment by ganeshmev** [comment]
Please revisit it.

**Comment by Ghostkeeper** [comment]
Then please fix the issues mentioned above. As far as I can tell, these issues are still open since November 2020:

* https://github.com/Ultimaker/Cura/pull/8650#issuecomment-727553939
* https://github.com/Ultimaker/Cura/pull/8650#issuecomment-727857347
* https://github.com/Ultimaker/Cura/pull/8650#pullrequestreview-532381666
* https://github.com/Ultimaker/Cura/pull/8650#issuecomment-728947203

**Pull request closed without merging**